### PR TITLE
Update dockerfile from xenial to bionic

### PR DESCRIPTION
**Pull request by tbille**
## Done

- upgrade from xenial to bionic
- upgrade node from 6.X to 10.X
- Fix bash var defined or not in `start-server`

## QA

- `./run build`
- `./run yarn run build-all`
- `docker build --build-arg COMMIT_ID=dfgd --tag tutorials . && docker run -ti -p 8016:80 tutorials`
- http://0.0.0.0:8016
- go on the website and make sure functionnalities still work

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-    mv /usr/lib/nodejs/node-v6.11.3-linux-x64 /usr/lib/nodejs/node-v6.11.3
-
-# Set nodejs paths
-ENV NODEJS_HOME=/usr/lib/nodejs/node-v6.11.3
-ENV PATH=$NODEJS_HOME/bin:$PATH
+RUN apt-get update && \
+    apt-get install curl xz-utils --yes && \
+    curl -sL https://deb.nodesource.com/setup_10.x | bash - && \
+    apt-get update && \
+    apt-get install --yes nodejs
```
Comment (nottrobin) [inline Dockerfile]: Can't we just move `curl xz-utils` here as well, so we don't need to do `apt-get install` twice?

File: package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "serve": "npm-run-all 'serve-examples {@}' --",
     "serve-examples": "./bin/serve $(if [ -z \"$*\" ]; then echo \"examples\"; fi)",
     "serve-live": "./bin/serve",
-    "start-server": "node ./server.js --root build/ --host 0.0.0.0 --port ${PORT:-80} $(if [ -n \"${BOT_PROXY:-}\" ]; then echo \"--bot-proxy ${BOT_PROXY}\"; fi)",
+    "start-server": "node ./server.js --root build/ --host 0.0.0.0 --port ${PORT:-80} $(if [ -z \"${BOT_PROXY:-}\" ]; then echo \"--bot-proxy ${BOT_PROXY}\"; fi)",
```
Comment (nottrobin) [inline package.json]: `-z` means "is empty". So what you're doing here is saying, if and only if `$BOT_PROXY` *doesn't* exist, should we add the `--bot-proxy` argument. Which doesn't make any sense. Unless I've misunderstood something.

File: Dockerfile (hunk 1 above)
Comment (tbille) [reply Dockerfile]: I still need to install curl to be able to install nodejs

Comment (nottrobin) [reply Dockerfile]: Oh lol. Yes you're right =)

File: package.json (hunk 2 above)
Comment (tbille) [reply package.json]: well here it didn't work for me by running with -n it was still passing the argunment

Comment (nottrobin) [reply package.json]: @tbille so I would suggest that probably means that the `--bot-proxy` argument is fundamentally broken, or maybe the value of the `${BOT_PROXY}` variable is wrong. But the fix isn't to simply reverse the conditional to something that makes no sense. The fix *may* be to remove `--bot-proxy` altogether, but only if you understand what you're doing.

Comment (tbille) [reply package.json]: ok when the var is defined but empty it still works

Comment (tbille) [reply package.json]:
```
$ BOT_PROXY=""
$ if [ -n ${BOT_PROXY:-} ]; then echo \"--bot-proxy ${BOT_PROXY}\"; fi
--bot-proxy
```

Comment (nottrobin) [reply package.json]: You need to wrap `${BOT_PROXY:-}` in quotes (`"${BOT_PROXY:-}"`) or it won't behave as you expect:

``` bash
$ [ -n  ] && echo true || echo false
true
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -20,11 +20,11 @@ jobs:
           command: yarn run build-all
       - run:
           name: Run the site server
-          command: PORT=80 ./bin/serve
+          command: PORT=8016 ./bin/serve
           background: true
       - run:
           name: Check site is accessible
-          command: sleep 4 && curl --head --fail --retry-delay 5 --retry 10  --retry-connrefused http://localhost
+          command: sleep 4 && curl --head --fail --retry-delay 5 --retry 10  --retry-connrefused http://localhost:8016
```
Comment (nottrobin) [inline .circleci/config.yml]: Why did you do this? I normally test all these sites on `80` to be more production-like.

File: .circleci/config.yml [hunk 4]
```diff
@@ -20,11 +20,11 @@ jobs:
           command: yarn run build-all
       - run:
           name: Run the site server
-          command: PORT=80 ./bin/serve
+          command: PORT=8016 ./bin/serve
```
Comment (nottrobin) [inline .circleci/config.yml]: Maybe this should instead be `yarn run build-all` followed-by `yarn run start-server` to be more production-like. Then at least it hopefully would have caught this error before @anthonydillon tried to release it, if it happened again.

**Comment by nottrobin** [comment]
Works a charm - QA +1 - but a couple of comments about the `test-site` CI step.

**Pull request merged**